src/service/resources-impl.js

```diff
@@ -1723,7 +1723,7 @@ export class Resource {
       return Promise.resolve();
     }
 
-    if (!this.renderOutsideViewport() && !this.isInViewport()) {
+    if (!this.isInViewport() && !this.renderOutsideViewport()) {
```

Comment:
ha. I had a TODO to explore having `prerenderOutsideViewport` method similar to renderOutsideViewport. For now I'm using `renderOutsideViewport` for both cases (prerender and layout). But yeah we might want to consider to separate these two.